File: indivs/20012024/2.py

def sum_of_x(a, b, my_func):
    # return get_square(a) + get_square(b)
    return my_func(a) + my_func(b)


res = sum_of_x(2, 5, lambda x: x*x)
res2 = sum_of_x(2, 2, lambda x: x*x*x)
# map, sorted, filter
print(res)
print(res2)

File: 5_module_frontend/topic_2/pizza_task/2.js
function my_func(sum = 0) {
  function inner(value) {
    if (value !== undefined) {
      sum += value;
      return inner;
    } else {
      return sum;
    }
  }
  return inner;
}

let res1 = my_func(1)(4)(1)(6)(); // res = 12
console.log(res1);
let res2 = my_func(1)(6)(); // res = 7
console.log(res2);
